src/components/synapse.rs

```diff
@@ -2,11 +2,27 @@ use std::{collections::HashMap, time::SystemTime};
 
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 
-use crate::api::routes::{
-    synapse::room_events::{FriendshipEvent, RoomEventRequestBody, RoomEventResponse},
-    v1::error::CommonError,
+use crate::{
+    api::routes::{
+        synapse::room_events::{RoomEventRequestBody, RoomEventResponse},
+        v1::error::CommonError,
```

Comment:
Ouch, yes, I'm very aware of it 🫠. However, if I were to try to abstract these errors to make them generic and then map the API and WebSocket errors accordingly, it would be a chaos. I think we can address all this in other PR as part of the open ticket we have.